test/compilable/issue20705.d

```diff
@@ -0,0 +1,12 @@
+// REQUIRED_ARGS: -preview=rvaluerefparam
+struct Foo
+{
+    int a;
+}
+
+void bar (T) (const ref T arg) {}
+
+void main ()
+{
+    bar(Foo(42));
+}
```

Comment:
```suggestion
// REQUIRED_ARGS: -preview=rvaluerefparam
struct Foo
{
    int[] a;
}

void bar (T) (const ref T arg) {}
T foo (T) (ref T arg) { return arg; }

void main ()
{
    bar(Foo([42]));
    auto x = foo(Bar([42]));
    static assert(is(typeof(x) == const(Foo)));
}
```

I think that's what @12345swordy has in mind. What's the type of a rvalue `ref`, `T` itself or `const T` ?
I think having it `const T` will be surprising to users, but it also matches the feedback in the DIP.

Reply:
this is a bit mangled, there is no `Bar`, and `bar` returns `void`. Can you redo  the commit suggestion please?

Reply:
My bad. It should've been `Foo`.